scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -519,6 +529,128 @@ impl ReplicatorConsumer {
         Ok(())
     }
 
+    fn delete_row_range_left(&mut self, data: CDCRow<'_>, included: bool) {
+        let (_, _, values) = self.get_common_cdc_row_data(&data);
+        self.left_range_included = included;
+        self.left_range_values = values[self.table_schema.partition_key.len()..]
+            .iter()
+            .map(|el| el.map(|x| x.clone()))
+            .collect();
```

Comment:
This seems to be a bit roundabout way to obtain a vector of clustering key values. The `get_common_cdc_row_data` computes several values but you only need one, moreover it creates a vector of pk + ck values, you discard the pks and clone the rest. Why not create a vector of cks manually and avoid clones altogether:

```rust
self.left_range_values = self
    .table_schema
    .clustering_key
    .iter()
    .map(|ck| data.take_value(ck))
    .collect();
```

Reply:
You can probably do something like this in `delete_row_range_right`, too. Perhaps it's worth extracting to a method.

Reply:
Additionally, consider removing the null values from the vector. For range deletes, the cdc row will contain a prefix of non-null ck values and then only nulls. In this case, adding `.take_while(Option::is_some)` before `.collect()`. should do the trick. This might simplify things.